File: server/src/main/java/org/cleanas2/service/polling/WatchedFile.java
package org.cleanas2.service.polling;

import org.joda.time.DateTime;

import java.nio.file.Path;

/**
 * @author Andrew Backer {@literal awbacker@gmail.com / andrew.backer@powere2e.com}
 */
public class WatchedFile {
    public final Path file;
    public final String senderId;
    public final String receiverId;
    public WatchStatus status = WatchStatus.NEW;
    public int retries;
    public DateTime sendAt;

    public WatchedFile(Path f, WatchedDir parent) {
        this(f, parent.senderId, parent.receiverId);
    }

    public WatchedFile(Path f, String s, String r) {
        this.file = f;
        this.senderId = s;
        this.receiverId = r;
        this.sendAt = DateTime.now().plusSeconds(15);
    }

    @Override
    public String toString() {
        return String.format("{file=%s, sender=%s, receiver=%s}", file.toString(), senderId, receiverId);
    }

}

File: server/src/main/java/org/cleanas2/service/polling/WatchedFile.kt
package org.cleanas2.service.polling

import org.joda.time.DateTime

import java.nio.file.Path

/**
 * @author Andrew Backer awbacker@gmail.com / andrew.backer@powere2e.com
 */
class WatchedFile(val file: Path, val senderId: String, val receiverId: String) {
    var status = WatchStatus.NEW
    var retries: Int = 0
    var sendAt: DateTime

    constructor(f: Path, parent: WatchedDir) : this(f, parent.senderId, parent.receiverId) {}

    init {
        this.sendAt = DateTime.now().plusSeconds(15)
    }

    override fun toString(): String {
        return String.format("{file=%s, sender=%s, receiver=%s}", file.toString(), senderId, receiverId)
    }

}
